pkg/awsutils/awsutils.go

```diff
@@ -826,14 +826,38 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 			Tags:         convertTagsToSDKTags(tags),
 		},
 	}
+	var needIPs = numIPs
 
-	input := &ec2.CreateNetworkInterfaceInput{
-		Description:       aws.String(eniDescription),
-		Groups:            aws.StringSlice(cache.securityGroups.SortedList()),
-		SubnetId:          aws.String(cache.subnetID),
-		TagSpecifications: tagSpec,
+	ipLimit := cache.GetENIIPv4Limit()
+	if ipLimit < needIPs {
+		needIPs = ipLimit
 	}
+	// If we don't need any more IPs, don't create ENI. It should never be the case.
+	if needIPs < 1 {
```

Comment:
This is safe to remove now? since we have the below check and this function is called from AllocENI...

```
if resourcesToAllocate > 0 {
		eni, err := c.awsClient.AllocENI(c.useCustomNetworking, securityGroups, subnet, resourcesToAllocate)
....
```